pkg/acs/lib/acs2d/do2d.c

```diff
@@ -237,42 +237,42 @@ int Do2D (ACSInfo *acs2d, int extver) {
       buff[0] = '\0';
 
 	    if (acs2d->detector != MAMA_DETECTOR) {
-        sprintf(MsgText, "    readnoise =");
+        snprintf(MsgText, sizeof(MsgText), "    readnoise =");
 
         for (i=0; i < NAMPS-1; i++) {
           if (acs2d->readnoise[i] > 0) {
-            sprintf (buff, "%.5g,",acs2d->readnoise[i]);
+            snprintf (buff, sizeof(buff), "%.5g,",acs2d->readnoise[i]);
             strcat (MsgText, buff);
           }
         }
 
         if (acs2d->readnoise[NAMPS-1] > 0) {
-          sprintf(buff, "%.5g",acs2d->readnoise[NAMPS-1]);
+          snprintf(buff, sizeof(buff), "%.5g",acs2d->readnoise[NAMPS-1]);
           strcat (MsgText, buff);
         }
 
         trlmessage(MsgText);
 
-        sprintf(MsgText, "    gain =");
+        snprintf(MsgText, sizeof(MsgText), "    gain =");
         for (i=0; i < NAMPS-1; i++) {
           if (acs2d->atodgain[i] > 0) {
-            sprintf(buff, "%.5g,",acs2d->atodgain[i]);
+            snprintf(buff, sizeof(buff), "%.5g,",acs2d->atodgain[i]);
             strcat (MsgText, buff);
           }
         }
 
         if (acs2d->atodgain[NAMPS-1] > 0) {
-          sprintf(buff, "%.5g",acs2d->atodgain[NAMPS-1]);
+          snprintf(buff, sizeof(buff), "%.5g",acs2d->atodgain[NAMPS-1]);
           strcat (MsgText, buff);
         }
 
-        trlmessage(MsgText);
+        trlmessage("%s", MsgText);
```

Comment:
So, with changes like this, will calwf3 and calstis still work as-is, or they won't work until a follow-up PR ensues?

Reply:
With the variadic functions you'd want to call it like `trlmessage("%s", MsgText)`, because it's explicit/correct. There are drawbacks to calling it like `trlmessage(MsgText)`. It's dangerous because say you have `%%s` in the buffer, and vasprintf converts it to `%s`. Now you have an errant format, and it _will_ try to parse it.

The behavior is identical to `printf`. The compiler will throw a warning if you try to do `printf(MsgText)`. Anyway, if such a thing occurs it'll throw an error and end the program. I've only encountered it in one area of the code, and I temporarily corrected it.